pkg/controller/remotecluster/cache.go

```diff
@@ -0,0 +1,44 @@
+package remotecluster
+
+import (
+	"sync"
+
+	"github.com/oecp/rama/pkg/rcmanager"
+	"k8s.io/klog"
+)
+
+type Cache struct {
+	*sync.RWMutex
+	rcMgrMap map[string]*rcmanager.Manager
+}
+
+func NewCache() *Cache {
+	return &Cache{
+		RWMutex:  new(sync.RWMutex),
+		rcMgrMap: make(map[string]*rcmanager.Manager),
+	}
+}
+
+func (c *Cache) Get(clusterName string) (manager *rcmanager.Manager, exists bool) {
+	c.RWMutex.RLock()
+	defer c.RWMutex.RUnlock()
+	manager, exists = c.rcMgrMap[clusterName]
+	return
+}
+
+func (c *Cache) Set(clusterName string, manager *rcmanager.Manager) {
+	c.RWMutex.Lock()
+	defer c.RWMutex.Unlock()
+
+	c.rcMgrMap[clusterName] = manager
+}
+
+func (c *Cache) Del(clusterName string) {
+	c.RWMutex.Lock()
+	defer c.RWMutex.Unlock()
+	if rc, exists := c.rcMgrMap[clusterName]; exists {
+		klog.Infof("Delete cluster %v from cache", clusterName)
+		rc.Close()
```

Comment:
why deleting a cluster from cache will close it firstly? this sequential actions should be guaranteed by caller

Reply:
What do  you mean, bro? Is it bad close in this sequence?

Reply:
I mean if someone have to shut down a cluster, close it firstly and delete it from cache secondly.
The two actions mentioned above do not have to be put into a single action.